trade-finance-manager-api/src/v1/controllers/update-facilities.js

```diff
@@ -11,66 +11,47 @@ const updateFacilities = async (deal) => {
 
   const {
     submissionDate: dealSubmissionDate,
-  } = deal.dealSnapshot.details;
-
-  const facilities = [
-    ...modifiedDeal.dealSnapshot.bondTransactions.items,
-    ...modifiedDeal.dealSnapshot.loanTransactions.items,
-  ];
-
-  const bonds = [];
-  const loans = [];
-
-  let updatedCount = 0;
-
-  return new Promise((resolve) => {
-    facilities.forEach(async (facility) => {
-      const {
-        _id: facilityId,
-        facilityType,
-      } = facility;
-
-      const facilityGuaranteeDates = getGuaranteeDates(facility, dealSubmissionDate);
-
-      const facilityCurrencyConversion = await convertFacilityCurrency(facility, dealSubmissionDate);
-      const facilityExposurePeriod = await getFacilityExposurePeriod(facility);
-      const facilityPremiumSchedule = await getFacilityPremiumSchedule(
-        facility,
-        facilityExposurePeriod,
-        facilityGuaranteeDates,
-      );
-
-
-      // TODO
-      // exposure period is not in unit test
-      const facilityUpdate = {
-        ...facilityCurrencyConversion,
-        ...facilityExposurePeriod,
-        facilityGuaranteeDates,
-        riskProfile: DEFAULTS.FACILITY_RISK_PROFILE,
-        premiumSchedule: facilityPremiumSchedule,
-      };
-      const updatedFacility = await api.updateFacility(facilityId, facilityUpdate);
-
-      updatedCount += 1;
-
-      // update deal object to return in response
-      if (facilityType === 'bond') {
-        bonds.push(updatedFacility);
-      } else if (facilityType === 'loan') {
-        loans.push(updatedFacility);
-      }
-
-      if (updatedCount === facilities.length) {
-        modifiedDeal.dealSnapshot.bondTransactions.items = bonds;
-        modifiedDeal.dealSnapshot.loanTransactions.items = loans;
-
-        return resolve(modifiedDeal);
-      }
-      return modifiedDeal;
-    });
-  });
+  } = modifiedDeal;
+
+  modifiedDeal.facilities = await Promise.all(modifiedDeal.facilities.map(async (facility) => {
+    const {
+      _id: facilityId,
+    } = facility;
+
+    const facilityGuaranteeDates = getGuaranteeDates(facility, dealSubmissionDate);
+
+    const facilityCurrencyConversion = await convertFacilityCurrency(facility, dealSubmissionDate);
+    const facilityExposurePeriod = await getFacilityExposurePeriod(facility);
+    const facilityPremiumSchedule = await getFacilityPremiumSchedule(
+      facility,
+      facilityExposurePeriod,
+      facilityGuaranteeDates,
+    );
+
+    // TODO
+    // exposure period is not in unit test
+    const facilityUpdate = {
+      ...facilityCurrencyConversion,
+      ...facilityExposurePeriod,
+      facilityGuaranteeDates,
+      riskProfile: DEFAULTS.FACILITY_RISK_PROFILE,
+      premiumSchedule: facilityPremiumSchedule,
+    };
+
+    const updateFacilityResponse = await api.updateFacility(facilityId, facilityUpdate);
+
+    const updatedFacility = {
```

Comment:
Does updateFacilityResponse return the complete facility or just whats been updated?
Is there need to reconstruct it again?

Reply:
I'll look into this, can probably be simplified

Reply:
Simplified this a lot here: https://github.com/foundry4/dtfs2/pull/206/commits/0e3614fa177cdb4bfe50e8a184ebcfd495f698c2